source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -521,6 +527,16 @@ class CaresDnsResolverFactory : public DnsResolverFactory {
     }
     return std::make_shared<Network::DnsResolverImpl>(cares, dispatcher, resolvers);
   }
+
+  void cleanup() override {
+    // Cleanup c-ares library if initialized.
+    if (ares_library_initialized_) {
+      ares_library_cleanup();
```

Comment:
Can we move this cleanup to the ares factory destructor? That would feel a little cleaner.

Reply:
Can't do it in the destructor since factories are statically initialized and will race with whatever static clean-up happens in c-ares.

Reply:
Makes sense. Can you comment that the destructor is not called in deterministic order as the object is statically constructed.